KO: Translation for ko/beta/tutorials/generative/dcgan.ipynb

Haabibi opened the pull request:
Translation for tutorials/generative/dcgan.ipynb

Below is a link to the original english version:
[https://github.com/tensorflow/docs/tree/master/site/en/r2/tutorials/generative/dcgan.ipynb](https://github.com/tensorflow/docs/tree/master/site/en/r2/tutorials/generative/dcgan.ipynb)

Thanks!

Haabibi commented:
It's my first time contributing to the community, and I have just signed the CLA!

rickiepark commented:
@Haabibi Thank you for contributing.
We will review your work soon. :)

Choiuijin1125 commented on site/ko/beta/tutorials/generative/dcgan.ipynb:
```diff
@@ -0,0 +1,988 @@
+   "source": [
+    "## 생성적 적대 신경망(GANs)은 무엇인가요? \n",
+    "\n",
+    "[생성적 적대 신경망](https://arxiv.org/abs/1406.2661) (GANs)은 요즘 컴퓨터과학에서 가장 흥미로운 아이디어 중 하나입니다. 두개의 모델이 적대적인 과정을 통해 동시에 학습됩니다. *생성자* (\"예술가\")는 진짜 처럼 보이는 이미지를 생성하도록 배우는 와중에, *감별자* (\"예술비평가\")는 가짜의 이미지로 부터 진짜를 구별하게 되는 것을 배우게 됩니다..   \n",
```
Just add one space :) 
I suggest '컴퓨터과학' → '컴퓨터 과학', 
'진짜 처럼' → '진짜처럼', 
'이미지로 부터' → '이미지로부터'  
And for the same reason, I suggest '학습' → '훈련'
And there are two periods on the last sentence. :) '..' → '.'

Choiuijin1125 commented on site/ko/beta/tutorials/generative/dcgan.ipynb:
```diff
@@ -0,0 +1,988 @@
+    "\n",
+    "\n",
+    "\n",
+    "학습하는 도중에, *생성자*는 점차 실제같은 이미지를 더 잘 생성하게 되면서, *감별자*는 점차 진짜와 가짜를 더 잘 구별하게됩니다. 이 과정은 *감별자*가 가짜 이미지에서 진짜 이미지를 더이상 구별하지 못하게 될때, 평형상태에 도달하게 됩니다. \n",
```
For the same reason, '학습' → '훈련'.
And I think, It's more natural '학습하는 도중에', → '훈련과정 동안' :)
And I suggest '되면서' → '되고', to emphasize the adversarial process.

Choiuijin1125 commented on site/ko/beta/tutorials/generative/dcgan.ipynb:
```diff
@@ -0,0 +1,988 @@
+    "\n",
+    "![생성자와 감별자를 그린 두번째 도표](https://tensorflow.org/alpha/tutorials/generative/images/gan2.png)\n",
+    "\n",
+    "이 노트북은 이 과정을 MNIST 데이터를 이용하여 보여줍니다. 아래의 애니메이션은 50 에폭(epoch)동안 훈련한 *생성자*가 생성해낸 연속된 이미지들을 보여줍니다. 이미지들은 랜덤한 잡음으로 부터 시작되었고, 점차적으로 시간이 지남에 따라 손으로 쓴 숫자들을 닮아가게 됩니다.  \n",
```
We are translating 'epoch' as '에포크', So I just suggest '에포크'. 
And I think It's better don't use Japanese char(的), If we can. 
So I suggest '점차적으로' → '점차'

rickiepark replied:
점차的으로 시간이 지남에 따라 --> 점차 시간이 지남에 따라 
:)

Haabibi replied:
it's very weird that 的 isn't showing on my screen.. 
I have made some changes on this line in new commits! :)

rickiepark replied:
I'm sorry. I don't mean you can see 的 char.
@Choiuijin1125 just pointed to '적' in '점차적으로'.
You already fixed it. :)

Choiuijin1125 commented on site/ko/beta/tutorials/generative/dcgan.ipynb:
```diff
@@ -0,0 +1,988 @@
+    "\n",
+    "![출력 예시](https://tensorflow.org/images/gan/dcgan.gif)\n",
+    "\n",
+    "생성적 적대 신경망 (GANs)에 대해 더 배우고 싶으시다면, MIT의 [Intro to Deep Learning](http://introtodeeplearning.com/) 수업을 추천합니다."
```
There is no reason do not translate this sentence. 
So I suggest '딥러닝 소개' :)

Choiuijin1125 commented on site/ko/beta/tutorials/generative/dcgan.ipynb:
```diff
@@ -0,0 +1,988 @@
+   "source": [
+    "### 생성자\n",
+    "\n",
+    "생성자는 시드값 (seed; 랜덤한 잡음)으로 부터 이미지를 생성하기 위해, `tf.keras.layers.Conv2DTranspose` (업 샘플링) 층들을 이용합니다. 이 시드값을 인풋으로 받는 `Dense`층으로 부터 시작을 하고, 희망했던 28x28x1 사이즈의 이미지가 나오도록 업샘플링을 여러번 합니다. tanh를  사용하는 마지막 층을 제외한 나머지 각 층마다 활성함수로 `tf.keras.layers.LeakyReLU`을 사용하고 있음을 주목합시다."
```
'으로 부터' → '으로부터', 
And I just suggest '인풋' → '입력' 
And For consistency, I suggest '업 샘플링' → '업샘플링' :)

Choiuijin1125 commented on site/ko/beta/tutorials/generative/dcgan.ipynb:
```diff
@@ -0,0 +1,988 @@
+   },
+   "source": [
+    "### 감별자 \n",
+    "감별자는 CNN 기반인 이미지 분류기입니다. "
```
CNN is mentioned for the first time in this file, 
So I suggest '합성곱 신경망(CNN, Convolutional Neural Network)' For a newbie like me :)

Choiuijin1125 commented on site/ko/beta/tutorials/generative/dcgan.ipynb:
```diff
@@ -0,0 +1,988 @@
+   },
+   "outputs": [],
+   "source": [
+    "# 이 메소드는 크로스 엔트로피 손실함수 (cross entropy loss)를 계산하기 위해 헬퍼 (helper) 함수를 반환합니다.\n",
```
We are mostly translating method as '메서드' 
So for consistency, I just suggest '메소드' → '메서드' :)

Choiuijin1125 commented on site/ko/beta/tutorials/generative/dcgan.ipynb:
```diff
@@ -0,0 +1,988 @@
+    "id": "0FMYgY_mPfTi"
+   },
+   "source": [
+    "## 손실함수와 최적함수 정의\n",
```
We are translating 'optimizer' as '옵티마이저'
So I suggest '최적함수' → '옵티마이저' :)

Choiuijin1125 commented on site/ko/beta/tutorials/generative/dcgan.ipynb:
```diff
@@ -0,0 +1,988 @@
+   },
+   "source": [
+    "## 손실함수와 최적함수 정의\n",
+    "두 모델의 손실함수와 최적함수를 정의합니다. "
```
For the same reason, '최적함수' → '옵티마이저'

Choiuijin1125 commented on site/ko/beta/tutorials/generative/dcgan.ipynb:
```diff
@@ -0,0 +1,988 @@
+    "id": "MgIc7i0th_Iu"
+   },
+   "source": [
+    "감별자와 생성자는 따로 학습되어지기 때문에, 감별자와 생성자의 최적함수 다릅니다."
```
For the same reason, '학습' → '훈련',
'최적함수' → '옵티마이저' :)

Choiuijin1125 commented on site/ko/beta/tutorials/generative/dcgan.ipynb:
```diff
@@ -0,0 +1,988 @@
+   },
+   "source": [
+    "### 체크포인트 저장\n",
+    "이 노트북은 오랫동안 진행되는 학습이 방해되는 경우에 유용할 수 있는 모델의 저장방법과 복구방법을 보여줍니다. "
```
For the same reason, '학습' → '훈련' :)

Choiuijin1125 commented on site/ko/beta/tutorials/generative/dcgan.ipynb:
```diff
@@ -0,0 +1,988 @@
+    "noise_dim = 100\n",
+    "num_examples_to_generate = 16\n",
+    "\n",
+    "# 우리는 이 시드를 시간이 지나도 재활용하겠습니다. \n",
```
Using '우리' is not natural in public document using Korean. 
So, I just suggest del '우리' :)

Choiuijin1125 commented on site/ko/beta/tutorials/generative/dcgan.ipynb:
```diff
@@ -0,0 +1,988 @@
+    "id": "jylSonrqSWfi"
+   },
+   "source": [
+    "훈련 루프는 생성자가 입력으로 랜덤시드를 받는 것으로 부터 시작됩니다. 그 시드값은 이미지를 생성하는데 사용되어 집니다. 감별자는 (훈련 세트에서 갖고온) 진짜 이미지와 (생성자가 생성해낸) 가짜이미지를 분류하기 위해 사용되어집니다. 각각의 모델들의 손실이 계산되어지고, 경사 (gradients)는 생성자와 감별자를 업데이트하기 위해 사용되어집니다."
```
'것으로 부터' → '것으로부터' :)

rickiepark replied:
For same reason above, I suggest '되어 집니다' --> '됩니다'
To make better sentence, you can translate like '그 시드값을 사용하여 이미지를 생성합니다' :)

I strongly suggest '감별자는 ~ 갖고온 ~ 생성해낸 ~ 분류하기 위해 사용되어집니다' --> '감별자를 사용하여 ~ **갖고 온** ~ **생성해 낸** ~ 분류합니다'

We don't translate 'gradient' becuase it's just not a slope. Please don't repeat '의' for easy reading.
I suggest '각각의 모델들의 손실이 계산되어지고, 경사 (gradients)는 생성자와 감별자를 업데이트하기 위해 사용되어집니다'
--> '**각 모델의** 손실**을** **계산하고**, **그래디언트(gradients)를 사용해** 생성자와 감별자를 업데이트**합니다**'

Haabibi replied:
oh wow! thank you so much for your kind comments. I now get a sense how I can translate better.

Choiuijin1125 commented on site/ko/beta/tutorials/generative/dcgan.ipynb:
```diff
@@ -0,0 +1,988 @@
+    "                             epoch + 1,\n",
+    "                             seed)\n",
+    "\n",
+    "    # 15 에폭이 지날 때마다 모델을 저장합니다.\n",
```
For the same reason, '에폭' → '에포크' :)

Choiuijin1125 commented on site/ko/beta/tutorials/generative/dcgan.ipynb:
```diff
@@ -0,0 +1,988 @@
+    "    if (epoch + 1) % 15 == 0:\n",
+    "      checkpoint.save(file_prefix = checkpoint_prefix)\n",
+    "\n",
+    "    print ('Time for epoch {} is {} sec'.format(epoch + 1, time.time()-start))\n",
```
Please translate print function :)

Haabibi replied:
I have added the below line for the translation of the print function, but I am not sure if this is the best expression.
# 출력: 에포크 (epoch+1)에서 걸린 시간은 (현재 시간(time.time())-start)초 입니다.

cre8tor replied:
I suggest,
print (' 에포크 {} 에서 걸린 시간은 {} 초 입니다'.format(epoch +1, time.time()-start))\n",
because python print() is actually functioning programming code.

Choiuijin1125 commented on site/ko/beta/tutorials/generative/dcgan.ipynb:
```diff
@@ -0,0 +1,988 @@
+    "\n",
+    "    print ('Time for epoch {} is {} sec'.format(epoch + 1, time.time()-start))\n",
+    "\n",
+    "  # 마지막 에폭이 끝난 후 생성합니다.\n",
```
'에폭' → '에포크' :)

Choiuijin1125 commented on site/ko/beta/tutorials/generative/dcgan.ipynb:
```diff
@@ -0,0 +1,988 @@
+   },
+   "source": [
+    "## 모델 학습\n",
+    "위에 정의된 `train()` 메소드를 생성자와 감별자를 동시에 학습하기 위해 부릅니다. 생성적 적대 신경망을 학습하는 것은 매우 까다로울 수 있습니다. 생성자와 감별자가 서로를 제압하지 않는 것이 중요합니다. (예시로 같은 비율로 학습하는 경우.)\n",
```
For the same reason, '메소드' → '메서드', '학습' → '훈련'. :)

Choiuijin1125 commented on site/ko/beta/tutorials/generative/dcgan.ipynb:
```diff
@@ -0,0 +1,988 @@
+   },
+   "outputs": [],
+   "source": [
+    "# 에폭 숫자를 사용하여 하나의 이미지를 보여줍니다.\n",
```
'에폭' → '에포크' :)

Choiuijin1125 commented on site/ko/beta/tutorials/generative/dcgan.ipynb:
```diff
@@ -0,0 +1,988 @@
+   "source": [
+    "## 모델 학습\n",
+    "위에 정의된 `train()` 메소드를 생성자와 감별자를 동시에 학습하기 위해 부릅니다. 생성적 적대 신경망을 학습하는 것은 매우 까다로울 수 있습니다. 생성자와 감별자가 서로를 제압하지 않는 것이 중요합니다. (예시로 같은 비율로 학습하는 경우.)\n",
+    "학습 초반부에는 생성된 이미지는 랜덤한 노이즈처럼 보입니다. 학습이 되어질 수록, 생성된 숫자들은 점차 진짜처럼 보여집니다. 약 50 에폭이 지난 후, 생성된 이미지들은 MNIST 숫자들처럼 닮아질것입니다. 코랩에서 기본 설정으로 돌릴 때, 에폭당 1분정도 소요될 것입니다."
```
'학습' → '훈련', '에폭' → '에포크' :)

Choiuijin1125 commented on site/ko/beta/tutorials/generative/dcgan.ipynb:
```diff
@@ -0,0 +1,988 @@
+    "id": "dZrd4CdjR-Fp"
+   },
+   "source": [
+    "## 모델 학습\n",
```
'모델 학습' → '모델 훈련' :)

rickiepark commented:
Hi @Haabibi ,
Please check the reviewer's suggestions?
Thanks!

Haabibi commented:
Sorry for the delay, I have addressed the comments. It would be great if reviewers can take a look at the comment on translating a print function on Line 707. Thank you!

rickiepark commented:
Thank you @Haabibi ,
I will add some comments in a few days. :)

rickiepark commented on site/ko/beta/tutorials/generative/dcgan.ipynb:
```diff
@@ -0,0 +1,989 @@
+   },
+   "outputs": [],
+   "source": [
+    "# `tf.function`이 어떻게 사용되어지는지 주목해 주세요.\n",
```
Please change '사용되어지는지' --> '사용되는지'

rickiepark commented on site/ko/beta/tutorials/generative/dcgan.ipynb:
```diff
@@ -0,0 +1,989 @@
+   "outputs": [],
+   "source": [
+    "# `tf.function`이 어떻게 사용되어지는지 주목해 주세요.\n",
+    "# 이 주석은 함수가 \"컴파일\"되어 지도록 합니다.\n",
```
'This annotation' is @tf.function decorator, not comment.
I suggest '이 주석은 함수가 \"컴파일\"되어 지도록 합니다' --> '이 데코레이터는 함수를 \"컴파일\"합니다'
Actually the decorator use jit compilation to make TF computation graph.

rickiepark commented on site/ko/beta/tutorials/generative/dcgan.ipynb:
```diff
@@ -0,0 +1,989 @@
+    "                             epoch + 1,\n",
+    "                             seed)\n",
+    "\n",
+    "    # 에포크 15개가 지날 때마다 모델을 저장합니다.\n",
```
'에포크 15개가' --> '에포크 15번이' or '15 에포크가' :)

rickiepark commented on site/ko/beta/tutorials/generative/dcgan.ipynb:
```diff
@@ -0,0 +1,989 @@
+   "source": [
+    "def generate_and_save_images(model, epoch, test_input):\n",
+    "  # `training`이 False로 맞춰진 것을 주목하세요.\n",
+    "  # 이렇게 하면 모든 층들이 추론 모드 (배치정규화)로 실행되어집니다. \n",
```
Because BatchNorm's parameter is modified by input data, it must be training=False in inference mode.
I suggest '이렇게 하면 모든 층들이 추론 모드 (배치정규화)로 실행되어집니다' --> '이렇게 하면 **(배치정규화를 포함하여)** 모든 층들이 추론 모드로 **실행됩니다**'

rickiepark commented on site/ko/beta/tutorials/generative/dcgan.ipynb:
```diff
@@ -0,0 +1,989 @@
+   },
+   "source": [
+    "## 모델 훈련\n",
+    "위에 정의된 `train()` 메서드를 생성자와 감별자를 동시에 훈련하기 위해 부릅니다. 생성적 적대 신경망을 학습하는 것은 매우 까다로울 수 있습니다. 생성자와 감별자가 서로를 제압하지 않는 것이 중요합니다. (예시로 같은 비율로 학습하는 경우.)\n",
```
'call' is often translated as '호출'.
I suggest '부릅니다' --> '호출합니다'

'similar rate' means speed.
I suggest '(예시로 같은 비율로 학습하는 경우.)' --> '(예를 들면 훈련 속도가 비슷해야 합니다)'

Please add one more a return char for line break.

Haabibi replied:
The original sentence in the tutorial is the following:
`It's important that the generator and discriminator do not overpower each other (e.g., that they train at a similar rate).`
As far as I know, the learning rate should be different to avoid G and D overpowering each other. What I have understood from this sentence was, if G and D are trained at a similar rate, they will overpower each other. 
So translation inside parentheses should entail such info. 
How about (예를 들면 훈련 속도가 비슷하면 서로 제압하게 됩니다.)

rickiepark replied:
Oh, I miss it. You are right. 
I just want to translate 'rate' as '학습률' and avoid repeated '제압'. :)
How about "(예를 들어 학습률이 비슷하면 한쪽이 우세해집니다.)"?

Haabibi replied:
sounds good to me!
i will change accordingly :)

rickiepark commented on site/ko/beta/tutorials/generative/dcgan.ipynb:
```diff
@@ -0,0 +1,989 @@
+   "source": [
+    "## 모델 훈련\n",
+    "위에 정의된 `train()` 메서드를 생성자와 감별자를 동시에 훈련하기 위해 부릅니다. 생성적 적대 신경망을 학습하는 것은 매우 까다로울 수 있습니다. 생성자와 감별자가 서로를 제압하지 않는 것이 중요합니다. (예시로 같은 비율로 학습하는 경우.)\n",
+    "훈련 초반부에는 생성된 이미지는 랜덤한 노이즈처럼 보입니다. 학습이 되어질 수록, 생성된 숫자들은 점차 진짜처럼 보여집니다. 약 50 에포크가 지난 후, 생성된 이미지들은 MNIST 숫자들처럼 닮아질것입니다. 코랩에서 기본 설정으로 돌릴 때, 에폭당 1분정도 소요될 것입니다."
```
I suggest '학습이 되어질 수록, 생성된 숫자들은 점차 진짜처럼 보여집니다' --> '**훈련이 진행될수록**, 생성된 **숫자는** 점차 진짜처럼 **보일 것입니다**'.
'생성된 이미지들은 MNIST 숫자들처럼 닮아질것입니다' --> 'MNIST 숫자와 닯은 이미지가 생성됩니다'
'돌릴 때, 에폭당' --> '실행하면, 에포크마다'

rickiepark commented on site/ko/beta/tutorials/generative/dcgan.ipynb:
```diff
@@ -0,0 +1,989 @@
+    "id": "NywiH3nL8guF"
+   },
+   "source": [
+    "`imageio`를 사용하여 훈련 중에 저장된 이미지들을 사용해 GIF 애니메이션을 만듭니다."
```
Repeated '사용' is hard to reading.
I suggest '`imageio`를 사용하여 훈련 중에 저장된 이미지들을 사용해' --> '**`imageio`로** 훈련 중에 저장된 **이미지를** 사용해'

rickiepark commented on site/ko/beta/tutorials/generative/dcgan.ipynb:
```diff
@@ -0,0 +1,989 @@
+    "id": "cGhC3-fMWSwl"
+   },
+   "source": [
+    "코랩에서 작업하고 계신다면, 아래의 코드에서 애니메이션을 다운로드 받을 수 있습니다: "
```
Please don't use honorific form.
'작업하고 계신다면' --> '작업하고 있다면'

Haabibi commented:
Thank you reviewers for reviewing this document! I really appreciate all of your comments! I have left one concern on Line 760, but other than that it seems like we are almost at the end for translating this document!

rickiepark commented:
Thank you so much @Haabibi 
LGTM 👍
